### ISSUE TITLE
Support for host name in file URI

### ISSUE THREAD

**Issue by biologghe** [jic-dtool/dtoolcore#7]

**Comment by tjelvar-olsson** [comment]
Could be implemented using by updating https://github.com/jic-dtool/dtoolcore/blob/56064ec31d274d72e60fa25f0c13dbc0744c4e12/dtoolcore/storagebroker.py#L410 to

```
return "{}://{}{}".format(cls.key, socket.gethostname(), dataset_abspath) 
```

Suggested that the host name should only be used for informative purposes, i.e. not used to validate that the hostname of the local machine matches the hostname in the URI.

Need to check that the fix works on Windows as well as Mac/Linux.

**Comment by tjelvar-olsson** [comment]
Fixed in 4ded884

**Comment by tjelvar-olsson** [comment]
Available in the 3.12.0 release of dtoolcore 3bf2c47
